install-test-gpu.cfg

```diff
@@ -38,10 +27,8 @@ write_files:
       User=root
       Type=oneshot
       RemainAfterExit=true
-      EnvironmentFile=/etc/nvidia-installer-env
-      ExecStartPre=/bin/bash -c 'mkdir -p "${NVIDIA_INSTALL_DIR_HOST}" && mount --bind "${NVIDIA_INSTALL_DIR_HOST}" "${NVIDIA_INSTALL_DIR_HOST}" && mount -o remount,exec "${NVIDIA_INSTALL_DIR_HOST}"'
-      ExecStart=/usr/bin/docker run --privileged --net=host --pid=host --volume "${NVIDIA_INSTALL_DIR_HOST}":"${NVIDIA_INSTALL_DIR_CONTAINER}" --volume /dev:/dev --volume "/":"${ROOT_MOUNT_DIR}" --env-file /etc/nvidia-installer-env "${COS_NVIDIA_INSTALLER_CONTAINER}"
-      ExecStartPost=/bin/bash -c "${NVIDIA_INSTALL_DIR_HOST}/bin/nvidia-smi"
+      ExecStartPre=/bin/bash -c "/usr/share/google/get_metadata_value attributes/run-installer-script > /tmp/run_installer.sh && chmod 755 /tmp/run_installer.sh"
+      ExecStart=/bin/bash /tmp/run_installer.sh
```

Comment:
You can move `chmod 755 /tmp/run_installer.sh` into separate ExecStartPre= command.
Then you don't really need the /bin/bash -c ".." either.

Reply:
I verified that `chmod` is not needed as we run the script using /bin/bash in ExecStart. However /bin/bash is still needed here because stdout redirection.